lib/carousel_slider.dart

```diff
@@ -73,97 +76,121 @@ class CarouselSliderState extends State<CarouselSlider> with TickerProviderState
   void didUpdateWidget(CarouselSlider oldWidget) {
     carouselState.options = options;
     carouselState.itemCount = widget.itemCount;
+
+    // Using this new method to handle animations when state changes
+    handleAutoPlay();
+
+    // pagecontroller needed to be re-initialized to respond to state changes
+    // I'm not too sure how bad this is for performance but it works :D
+    pageController = PageController(
+      viewportFraction: options.viewportFraction,
+      initialPage: carouselState.realPage,
+    );
+
+    carouselState.pageController = pageController;
+
```

Comment:
Calling `handleAutoPlay` and re-initializing the `pageController` to respond to changes in state.